### PR TITLE
Modal flow changes

### PR DESCRIPTION
**Pull request by rperez89**
This address this issue `We should add a third step to the modal for the user to allow the 1Hive protocol` from #336

### CODE REVIEW

File: src/components/ModalFlows/StakeScreens/StakeScreens.js [hunk 1]
```diff
@@ -11,6 +11,8 @@ function StakeScreens({ mode, stakeManagement, stakeActions }) {
   const [transactions, setTransactions] = useState([])
   const { accountBalance } = useAppState()
 
+  const { allowance: stakingAllowance } = stakeManagement?.staking || null
```
Comment (fabriziovigevani) [inline src/components/ModalFlows/StakeScreens/StakeScreens.js:14]:
```suggestion
  const { allowance: stakingAllowance } = stakeManagement?.staking || {}
```

File: src/components/ModalFlows/StakeScreens/StakeScreens.js [hunk 2]
```diff
@@ -27,13 +29,27 @@ function StakeScreens({ mode, stakeManagement, stakeActions }) {
...
-          onComplete()
+          temporatyTrx.current = temporatyTrx.current.concat(intent)
+
+          if (stakingAllowance?.gt(0)) {
+            setTransactions(temporatyTrx.current)
+            onComplete()
+          }
         })
+
+        if (!stakingAllowance?.gt(0)) {
```
Comment (fabriziovigevani) [inline src/components/ModalFlows/StakeScreens/StakeScreens.js:42]:
```suggestion
        if (stakingAllowance?.eq(0)) {
```

File: src/hooks/useStaking.js [hunk 3]
```diff
@@ -334,17 +334,41 @@ export function useStaking() {
...
+          onDone(intent)
+        }
+        return
+      }
+
+      await stakingContract.allowManager(
+        connectedAgreementApp.address,
+        MAX_INT.toString(10),
+        '0x'
+      )
```
Comment (fabriziovigevani) [inline src/hooks/useStaking.js:368]: Even though it's the same lets put it inside the else and remove the return, might be a bit more clearer to read.
```suggestion
      } else {
         await stakingContract.allowManager(
            connectedAgreementApp.address,
            MAX_INT.toString(10),
            '0x'
          )
      }
```

**Review by fabriziovigevani — COMMENTED** [review]
Nice

Some minor comments and corrections

Edit: I will do the suggested changes on my PR with the integration and ifyou don't agree with some of my comments let me know in the PR

**Comment by rperez89** [comment]
As talked in discord going to merge this and the changes were made in your pr

**Pull request merged**